core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1574,36 +1616,52 @@ public void run() {
 
         @Override
         protected ManagedChannelBuilder<?> delegate() {
-          return managedChannelImplBuilder;
+          return delegate;
         }
 
         @Override
         public ManagedChannel build() {
           // TODO(creamsoup) prevent main channel to shutdown if oob channel is not terminated
-          return new ManagedChannelImpl(
-                  managedChannelImplBuilder,
-                  originalTransportFactory,
-                  backoffPolicyProvider,
-                  balancerRpcExecutorPool,
-                  stopwatchSupplier,
-                  Collections.<ClientInterceptor>emptyList(),
-                  timeProvider);
+          // TODO(zdapeng) register the channel as a subchannel of the parent channel in channelz.
+          if (managedChannelImplBuilder != null) {
+            return new ManagedChannelImpl(
```

Comment:
We discussed offline how to get rid of this case and always just delegate here. (Which means the build() method here can just be deleted.)

Reply:
Done.